editor/src/components/canvas/controls/select-mode/absolute-resize-control.tsx

```diff
@@ -0,0 +1,334 @@
+    const dispatch = useEditorState((store) => store.dispatch, 'ResizeEdge dispatch')
+    const jsxMetadataRef = useRefEditorState((store) => store.editor.jsxMetadata)
+    const selectedViewsRef = useRefEditorState((store) => store.editor.selectedViews)
+
+    const onEdgeMouseDown = React.useCallback(
+      (event: React.MouseEvent<HTMLDivElement>) => {
+        startResizeInteraction(
+          event,
+          dispatch,
+          props.position,
+          props.enabledDirection,
+          jsxMetadataRef.current,
+          selectedViewsRef.current,
+        )
+      },
+      [dispatch, props.position, props.enabledDirection, jsxMetadataRef, selectedViewsRef],
+    )
+
+    const lineSize = ResizeMouseAreaSize / scale
+    const width = props.direction === 'horizontal' ? undefined : lineSize
+    const height = props.direction === 'vertical' ? undefined : lineSize
+    const offsetLeft = props.direction === 'horizontal' ? undefined : -lineSize / 2
+    const offsetTop = props.direction === 'vertical' ? undefined : -lineSize / 2
+    return (
+      <div
+        style={{
+          position: 'absolute',
+          top: offsetTop,
+          left: offsetLeft,
+          pointerEvents: 'none',
+        }}
+      >
+        <div
+          ref={ref}
+          style={{
+            position: 'relative',
+            width: width,
+            height: height,
+            backgroundColor: 'transparent',
+            cursor: props.cursor,
+            pointerEvents: 'initial',
+          }}
+          onMouseDown={onEdgeMouseDown}
+        ></div>
+      </div>
+    )
+  }),
+)
+
+function startResizeInteraction(
+  event: React.MouseEvent<HTMLDivElement>,
+  dispatch: EditorDispatch,
+  position: EdgePosition,
+  enabledDirection: EdgePosition,
+  metadata: ElementInstanceMetadataMap,
+  selectedViews: Array<ElementPath>,
+) {
+  event.stopPropagation()
+  if (event.buttons === 1) {
+    // TODO update this to call createInteractionState to use canvas strategies
+    const centerBasedResize = event.altKey
+    const keepAspectRatio = event.shiftKey // || props.elementAspectRatioLocked ???
+    const enableSnapping = !event.metaKey
+    const canvasPositions = windowToCanvasCoordinatesGlobal(
+      windowPoint({ x: event.nativeEvent.x, y: event.nativeEvent.y }),
+    )
+    const start: CanvasPoint = canvasPositions.canvasPositionRaw
+    const originalFrames = getOriginalFrames(selectedViews, metadata)
+    const isMultiSelect = selectedViews.length !== 1
+
+    const originalSize = boundingRectangleArray(
+      selectedViews.map((path) => MetadataUtils.getFrameInCanvasCoords(path, metadata)),
+    )
+    let possibleTargetProperty: LayoutTargetableProp | undefined = undefined
+    if (position.x === 0.5) {
+      possibleTargetProperty = 'height'
+    } else if (position.y === 0.5) {
+      possibleTargetProperty = 'width'
+    }
+
+    if (originalSize != null) {
+      const newDragState = updateResizeDragState(
```

Comment:
I guess we need a follow on task from this to bring this into the canvas strategy world?

Reply:
yep, I updated the paper doc!